src/lib/core/datetime.c

```diff
@@ -789,6 +791,12 @@ datetime_increment_by(struct datetime *self, int direction,
 	if (rc != 0)
 		return rc;
 
+	if (tzindex != 0) {
+		int isdst = 0;
+		long gmtoff = offset * 60;
+		epoch_timezone_lookup(secs, tzindex, &gmtoff, &isdst);
+		offset = gmtoff / 60;
+	}
 	self->epoch = utc_secs(secs, offset);
 	self->nsec = nsec;
 	self->tzoffset = datetime_gmtoff(self) / 60;
```

Comment:
You can replace this with `self->tzoffset = offset` now, since you call `epoch_timezone_lookup` for non-null tzindex above

Reply:
Done so, thanks!
```diff
diff --git a/src/lib/core/datetime.c b/src/lib/core/datetime.c
index 3507b7c6a..e22504d2f 100644
--- a/src/lib/core/datetime.c
+++ b/src/lib/core/datetime.c
@@ -799,7 +799,7 @@ datetime_increment_by(struct datetime *self, int direction,
 	}
 	self->epoch = utc_secs(secs, offset);
 	self->nsec = nsec;
-	self->tzoffset = datetime_gmtoff(self) / 60;
+	self->tzoffset = offset;
 	return 0;
 }
```